src/SpaceKitProvider/index.tsx

```diff
@@ -47,12 +47,25 @@ const SpaceKitSetContext = React.createContext<
  */
 export const SpaceKitProvider: React.FC<Partial<State>> = ({
   children,
-  ...stateProps
+  theme,
+  disableAnimations,
 }) => {
-  const [state, setState] = React.useState<State | undefined>({
-    ...defaultState,
-    ...stateProps,
-  });
+  const nextState = useMemo(
+    () => ({
+      theme: theme ?? defaultState.theme,
+      disableAnimations: disableAnimations ?? defaultState.disableAnimations,
+      singletonComponents: {},
+    }),
+    [theme, disableAnimations],
+  );
+  const [state, setState] = React.useState<State>(nextState);
+
+  useEffect(() => {
+    setState((currState) => ({
+      ...nextState,
+      singletonComponents: currState?.singletonComponents,
```

Comment:
NBD we don't need the ? here :)
```suggestion
      singletonComponents: currState.singletonComponents,
```